packages/connectors/connector-dingtalk-web/src/index.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * DingTalk OAuth2 Connector
+ * https://open.dingtalk.com/document/orgapp/obtain-identity-credentials#title-4up-u8w-5ug
+ */
+
+import { assert } from '@silverhand/essentials';
+import { got, HTTPError } from 'got';
+
+import type {
+  GetAuthorizationUri,
+  GetUserInfo,
+  GetConnectorConfig,
+  CreateConnector,
+  SocialConnector,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  validateConfig,
+  ConnectorType,
+  parseJson,
+} from '@logto/connector-kit';
+
+import {
+  authorizationEndpoint,
+  accessTokenEndpoint,
+  userInfoEndpoint,
+  scope as defaultScope,
+  defaultMetadata,
+  defaultTimeout,
+} from './constant.js';
+import type { DingtalkConfig } from './types.js';
+import {
+  dingtalkConfigGuard,
+  accessTokenResponseGuard,
+  userInfoResponseGuard,
+  authResponseGuard,
+} from './types.js';
+
+const getAuthorizationUri =
+  (getConfig: GetConnectorConfig): GetAuthorizationUri =>
+  async ({ state, redirectUri }) => {
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, dingtalkConfigGuard);
+
+    const { clientId, scope } = config;
+
+    const queryParameters = new URLSearchParams({
+      client_id: clientId,
+      redirect_uri: encodeURI(redirectUri), // The variable `redirectUri` should match {clientId, clientSecret}
+      response_type: 'code',
+      scope: scope ?? defaultScope,
+      state,
+      prompt: 'consent',
+    });
+
+    return `${authorizationEndpoint}?${queryParameters.toString()}`;
+  };
+
+export const getAccessToken = async (
+  code: string,
+  config: DingtalkConfig
+): Promise<{ accessToken: string }> => {
+  const { clientId, clientSecret } = config;
+
+  const httpResponse = await got.post(accessTokenEndpoint, {
+    json: {
+      clientId,
+      clientSecret,
+      code,
+      grantType: 'authorization_code',
+    },
+    timeout: { request: defaultTimeout },
+  });
+
+  const result = accessTokenResponseGuard.safeParse(parseJson(httpResponse.body));
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+  }
+
+  const { accessToken } = result.data;
+  assert(accessToken, new ConnectorError(ConnectorErrorCodes.InvalidResponse));
```

Comment:
Why not make `accessToken` required in `accessTokenResponseGuard`? This way, first we don't need to manually check if `accessToken` exists, and second, if `accessToken` doesn't exist, we can get a more detailed error message.